fn_falcon_sandbox/fn_falcon_sandbox/components/falcon_sandbox_submit_file.py

```diff
@@ -0,0 +1,216 @@
+                )
+            else:
+                yield StatusMessage("> Function inputs OK")
+
+            falcon_sandbox_environment_id = get_environment_id(
+                falcon_sandbox_environment
+            )
+            falcon_sandbox_action_script = get_runtime_action_script(
+                falcon_sandbox_action_script
+            )
+
+            ## Making sure all param names are following naming conventions
+            for p in kwargs:
+                if not "falcon_sandbox_" in p:
+                    FunctionError(
+                        "Input parameter name must start with 'falcon_sandbox_'"
+                    )
+
+            client = self.rest_client()
+            time_now = str(time.time())
+            request_common = RequestsCommon(self.options, kwargs)
+
+            ## Get the file for submission
+            yield StatusMessage(
+                "Getting Attachment from Incident {}".format(falcon_sandbox_incident_id)
+            )
+
+            attachment, attachment_metadata = get_file_attachment_and_metadata(
+                client,
+                falcon_sandbox_incident_id,
+                falcon_sandbox_artifact_id,
+                falcon_sandbox_task_id,
+                falcon_sandbox_attachment_id,
+            )
+
+            attachment_path, temp_files = write_temp_file(
+                attachment, "attachment_{}".format(time_now)
+            )
+
+            attachment_name = attachment_metadata.get("name")
+            attachment_type = attachment_metadata["content_type"]
+
+            yield StatusMessage(
+                "Environment using for analysis: {}".format(falcon_sandbox_environment)
+            )
+
+            ## Prepare for submission
+            url, http_method = HA_REST_API_URLS.get("submit_file")
+            url = url.format(API_HOST)
+            form_data = {
+                "file": (attachment_name, open(attachment_path, "rb"), attachment_type)
+            }
+
+            for p in HA_LIST_OF_RUNTIME_PARAMS_SUBMIT_FILE:
+                k = p.split("falcon_sandbox_")[1]
+                v = eval(p)
+                form_data[k] = v
+
+            submit_payload = MultipartEncoder(form_data)
+
+            submit_header = falcon_sandbox_request_header(API_KEY)
+            submit_header["content-type"] = submit_payload.content_type
+
+            ## Submit file to falcon sandbox
+            yield StatusMessage("Submitting...")
+            response = request_common.execute_call(
+                http_method, url, submit_payload, log=log, headers=submit_header
+            )
+            job_id = response["job_id"]
+            url, http_method = HA_REST_API_URLS.get("get_state")
+            url = url.format(API_HOST, job_id)
+            yield StatusMessage("Successfully Submitted. \n Job ID: {}".format(job_id))
+            ## Get Status of Analysis
```

Comment:
* This code to 'Get Status of Analysis' seems to be very similar in the Submit File and Submit URL functions
* Could we add it to the `helper` to avoid repeated code?

Reply:
Yes this thought can in my mind but : 

- the code uses StatusMessage, FunctionError and a function should be as isolated as possible, so I believe a helper method shouldn't influence resilient function.
- In future one may want to handle reports of both File and URL differently .